src/Scheduler/StatefulScheduledCommandsResolver.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Scheduler;
+
+use DateTime;
+use Tempest\Console\ConsoleCommand;
+
+final class StatefulScheduledCommandsResolver implements ScheduledCommandsResolver
+{
+    public const string CACHE_PATH = __DIR__ . '/last-schedule-run.cache.php';
+
+    public function __construct(
+        private SchedulerConfig $config,
+    ) {
+    }
+
+    public function resolve(DateTime $date): array
+    {
+        $lastRuns = $this->get();
+
+        $eligibleToRun = array_filter(
+            $this->config->schedules,
+            fn (ConsoleCommand $command) => $this->isCommandEligible(
+                command: $command,
+                now: $date,
+                lastRunTimestamp: $lastRuns[$command->getName()] ?? null,
+            )
+        );
+
+        $this->markAsRan($eligibleToRun, $date);
+
+        return $eligibleToRun;
+    }
+
+    /**
+     * Returns a key value array of the last run time of each command.
+     * The key is the command name and the value is the last run time in unix timestamp.
+     *
+     * @return array<string, int>
+     */
+    public function get(): array
```

Comment:
Could these methods be private?

Reply:
I'll refactor the whole thing later on - the intend of this was to listen to feedback on how you'd want this to work high-level first 😎